src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidator.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import java.util.Optional;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.exception.ServiceException;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsWithinOneYear.CreditorsWithinOneYear;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsWithinOneYear.CurrentPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.creditorsWithinOneYear.PreviousPeriod;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.CompanyService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+
+@Component
+public class CreditorsWithinOneYearValidator extends BaseValidator {
+
+    @Value("${invalid.note}")
+    private String invalidNote;
+
+    private static final String CREDITORS_WITHIN_PATH = "$.creditors_within_one_year";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".current_period";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH = CREDITORS_WITHIN_PATH +
+            ".previous_period";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_PREVIOUS_PERIOD_TOTAL_PATH =
+            CREDITORS_WITHIN_PREVIOUS_PERIOD_PATH + ".total";
+    private static final String CREDITORS_WITHIN_CURRENT_PERIOD_DETAILS_PATH =
+            CREDITORS_WITHIN_CURRENT_PERIOD_PATH + ".details";
+
+    private CompanyService companyService;
+
+    @Autowired
+    public CreditorsWithinOneYearValidator(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    public Errors validateCreditorsWithinOneYear(@Valid CreditorsWithinOneYear creditorsWithinOneYear, Transaction transaction) throws DataException {
+
+        Errors errors = new Errors();
+
+        if (creditorsWithinOneYear != null) {
```

Comment:
We shouldn't ever need a null check on the resource. It _is_ better to program defensively, but the resource should exist by this point; when data is `POST`'ed or `PUT`, if Spring fails to map the JSON to an object and instead provides null we have bigger problems!